### ISSUE TITLE
Status of this crate?

### ISSUE THREAD

**Issue by devmattrick** [bouffalolab/bl808-pac#2]
I was excited to see that there was a Rust crate being developed for the BL808. I noticed that most of this crate is still TODO and I was wondering if there's any near term plans to continue development on this? If so, is there anything I can do to assist with this? Thanks!

**Comment by EnzoAlbornoz** [comment]
From what I saw on the Rust community, PAC crates are autogenerated through the SVD files (a description of the SoC memory map and features). Then a `bl808-hal` crate should implement the traits present on `embedded-hal`.

**Comment by devmattrick** [comment]
Yep, from my understanding we'd just need a complete [SVD file](https://github.com/bouffalolab/bl808-pac/blob/main/bl808.svd) for the chip. I don't really have any experience in this domain, though.

**Comment by luojia65** [comment]
The SVD file is translated from chip documents line by line, especially register definitions and fields. You can help with that if possible :)

**Comment by gamelaster** [comment]
@luojia65 isn't better to parse source code instead? Often I found wrong information in datasheet and reference manual, where mostly in SDK headers it was mostly correct.

**Comment by luojia65** [comment]
@gamelaster Thanks for mentioning! C langauge SDK headers are more reliable sources, while the datasheet may providers information of how the peripheral works overall.

**Comment by Codeleiche** [comment]
Hello, I'd like to help completing the [SVD file](https://github.com/bouffalolab/bl808-pac/blob/main/bl808.svd) but can't find any good source of the registers. Anyone from Bouffalolab who can point me to a reliable source? (I do not have a board I can validate the registers on that's why I'm picky) until now I found [this sdk](https://github.com/bouffalolab/bl_mcu_sdk) but it seams to be incomplete and [this documantation](https://github.com/bouffalolab/bl_docs) but it does not render....

**Comment by EnzoAlbornoz** [comment]
@Codeleiche 
> Hello, I'd like to help completing the [SVD file](https://github.com/bouffalolab/bl808-pac/blob/main/bl808.svd) but can't find any good source of the registers. Anyone from Bouffalolab who can point me to a reliable source? (I do not have a board I can validate the registers on that's why I'm picky) until now I found [this sdk](https://github.com/bouffalolab/bl_mcu_sdk) but it seams to be incomplete and [this documantation](https://github.com/bouffalolab/bl_docs) but it does not render....

I think the most reliable resource are the ones that you mentioned. I suggest using them to write the missing parts, focusing on `bl_mcu_sdk` (reference by implementation 🤷)
